src/Caching/Hybrid/src/Internal/DefaultHybridCache.StampedeStateT.cs

```diff
@@ -0,0 +1,268 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Microsoft.Extensions.Caching.Hybrid.Internal;
+
+partial class DefaultHybridCache
+{
+    internal sealed class StampedeState<TState, T> : StampedeState
```

Comment:
What are `TState` and `T`?  I think `T` might be the type of the thing in the cache?  I'm not sure I understand what `TState` is though, other than that it's not a stampede state.

Reply:
Maybe it's a way for the caller to stash data they'll want when some callback of theirs is invoked?

Reply:
Yes, that

Reply:
I might call `T` "TEntry" or something.  But I assume it's called "T" everywhere else in the caching code and that's why it's like that here?

Reply:
yes; the main "thing" here is the value of whatever we're fetching,  and in the stateless version there is *only* `T`; the *state* (`TState`) is incidental and secondary, and I don't think it needs additional disambiguation. The `T` and `TState` are documented in the public API